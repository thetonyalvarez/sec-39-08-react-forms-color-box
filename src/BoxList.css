.BoxList {
  max-width: 600px !important;
  margin: auto;
}

.BoxGrid .Box {
  display: inline-block;
  margin: 8px;
}