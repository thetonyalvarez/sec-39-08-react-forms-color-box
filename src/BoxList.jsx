import { useState } from 'react'
import { v4 as uuid } from 'uuid';
import Box from './Box'
import NewBoxForm from './NewBoxForm'
import { Box as BoxUI } from '@mui/material'
import Container from '@mui/material/Container'
import './BoxList.css'



const BoxList = () => {
  const [boxes, setBoxes] = useState([])
  
  // add new box
  const addBox = (box) => {
    let newBox = {...box, id: uuid()}
    setBoxes(boxes => [...boxes, newBox])
  }
  // end addBox

  // remove box
  const removeBox = (id) => {
    setBoxes(boxes => boxes.filter(box => box.id !== id))
  }
  // end box

  // render all boxes
  const renderBoxes = () => {
    return (
      <BoxUI>
        {boxes.map(box => (
          <Box
            key={box.id}
            id={box.id}
            bgColor={box.bgColor}
            width={box.width}
            height={box.height}
            handleRemove={removeBox}
          />
        ))}
      </BoxUI>
    )
  }

  return (
    <Container className="BoxList">
      <NewBoxForm addBox={addBox}/>
      {renderBoxes()}
    </Container>
  )
}

export default BoxList;