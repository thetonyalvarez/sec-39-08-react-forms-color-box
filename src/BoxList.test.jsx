import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import BoxList from './BoxList'

// smoke test
it("renders without crashing", function() {
  render(<BoxList />)
});

// snapshot
it("matches snapshot", function() {
  const {asFragment} = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});