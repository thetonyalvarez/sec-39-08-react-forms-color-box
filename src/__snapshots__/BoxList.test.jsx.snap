// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg BoxList css-1oqqzyl-MuiContainer-root"
  >
    <form
      autocomplete="off"
      class="MuiBox-root css-0"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="bgColor"
          id="bgColor-label"
        >
          Color
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="bgColor"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Color
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="width"
          id="width-label"
        >
          Width
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="width"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Width
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="height"
          id="height-label"
        >
          Height
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="height"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Height
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Add new box
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
    <div
      class="MuiBox-root css-0"
    />
  </div>
</DocumentFragment>
`;
